allennlp/commands/build_vocab.py

```diff
@@ -0,0 +1,100 @@
+"""
+Subcommand for building a vocabulary from a training config.
+"""
+
+import argparse
+import json
+import logging
+import os
+import tarfile
+import tempfile
+
+from overrides import overrides
+
+from allennlp.commands.subcommand import Subcommand
+from allennlp.common.params import Params
+from allennlp.training.util import make_vocab_from_params
+
+
+logger = logging.getLogger(__name__)
+
+
+@Subcommand.register("build-vocab")
+class BuildVocab(Subcommand):
+    @overrides
+    def add_subparser(self, parser: argparse._SubParsersAction) -> argparse.ArgumentParser:
+        description = """Build a vocabulary from an experiment config file."""
+        subparser = parser.add_parser(self.name, description=description, help=description)
+
+        subparser.add_argument("param_path", type=str, help="path to an experiment config file")
+
+        subparser.add_argument(
+            "output_path", type=str, help="path to save the vocab tar.gz file to"
+        )
+
+        subparser.add_argument(
+            "-f",
+            "--force",
+            action="store_true",
+            help="force write if the output_path already exists",
+        )
+
+        subparser.add_argument(
+            "-o",
+            "--overrides",
+            type=str,
+            default="",
+            help=(
+                "a json(net) structure used to override the experiment configuration, e.g., "
+                "'{\"iterator.batch_size\": 16}'.  Nested parameters can be specified either"
```

Comment:
You might consider a more meaningful help text here.
```suggestion
                "'{\"vocabulary.min_count\": {\"labels\": 10}}'.  Nested parameters can be specified either"
```

(not sure if using dots the whole way would work...  maybe it would)